src/httpserver.cpp

```diff
@@ -319,8 +315,8 @@ static bool HTTPBindAddresses(struct evhttp* http)
         LogPrintf("Binding RPC on address %s port %i\n", i->first, i->second);
         evhttp_bound_socket *bind_handle = evhttp_bind_socket_with_handle(http, i->first.empty() ? nullptr : i->first.c_str(), i->second);
         if (bind_handle) {
-            CNetAddr addr;
-            if (i->first.empty() || (LookupHost(i->first, addr, false) && addr.IsBindAny())) {
+            const std::optional<CNetAddr> addr{LookupHost(i->first, false)};
+            if (i->first.empty() || (addr.has_value() && addr->IsBindAny())) {
```

Comment:
nit: Seems like this is a slight change in the logic. On master, we don't call `LookupHost` if the address is an empty string, on the PR it is called unconditionally and only then the `empty()` check is done. The resulting behaviour is the same though, since LookupHost returns early anways if the passed address is empty. Could probably remove the `i->first.empty()` check here?

Reply:
Yes, you're right!